backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/paymentallocation/service/AllocationLineCandidateSaver.java

```diff
@@ -156,6 +158,11 @@ private PaymentAllocationId saveCandidate_InvoiceToInvoice(@NonNull final Alloca
 				.writeOffAmt(writeOffAmt)
 				.build());
 
+		final I_C_Invoice payableInvoice = candidate.getPayableDocumentRef().getModel(I_C_Invoice.class);
+		// Special case: Allocations of sales credit memos and purchase invoices must always be negated. See #20395
+		final boolean negateAllocationAmount = invoiceBL.isARCreditMemo(payableInvoice) || (!payableInvoice.isSOTrx() && !invoiceBL.isCreditMemo(payableInvoice));
```

Comment:
IMHO saver is not the right place to do this.
This class shall have one responsibility, to persist the allocation candidates and take any BL decisions.
The amounts shall come in the right shape before this.

Reply:
also pls note, if u are negating payamt, discount, write-off ... not sure if over/uder amount stays the same...

Reply:
Added a proto with todos where the amount is negated on allocation candidate creation. 
The only part needed in the saver is to not negate the service fee invoice allocation if the payAmt is already negative, to achieve the allocation in scope of this issue: 


![image](https://github.com/user-attachments/assets/a8699866-f6eb-4b1d-bfbc-0d53223f06bd)